models/gif_psc_exp.h

```diff
@@ -0,0 +1,426 @@
+#include "ring_buffer.h"
+#include "connection.h"
+#include "universal_data_logger.h"
+
+#include "nest.h"
+
+/* BeginDocumentation
+  Name: gif_psc_exp - Current based generalized integrate-and-fire neuron model
+  according to Mensi et al. (2012)
+  and Pozzorini et al. (2015)
+
+  Description:
+
+  gif_psc_exp is the generalized integrate-and-fire neuron according to Mensi et
+  al. (2012)
+  and Pozzorini et al. (2015), with exponential shaped postsynaptic currents.
+
+  This model features both an adaptation current and a dynamic threshold for
+  spike-frequency
+  adaptation. The membrane potential (V) is described by the differential
+  equation:
+
+  C*dV(t)/dt = -g_L*(V(t)-E_L) - eta_1(t) - eta_2(t) - ... - eta_n(t) + I(t)
+
+  where each eta_i is a spike triggered current (stc), and the neuron model can
+  have arbitrary number of them.
+  Dynamic of each eta_i is described by:
+
+  Tau_eta_i*d{eta_i}/dt = -eta_i
+
+  and in case of spike emission, its value increased by a constant (which can be
+  positive or negative):
+
+  eta_i = eta_i + q_eta_i  (in case of spike emission).
+
+  Neuron produces spikes STOCHASTICALLY according to a point process with the
+  firing intensity:
+
+  lambda(t) = lambda_0 * exp[(V(t)-V_T(t)/Delta_V)]
+
+  where V_T(t) is a time-dependent firing threshold:
+
+  V_T(t) = V_T_star + gamma_1(t) + gamma_2(t) + ... + gamma_m(t)
+
+  where gamma_i is a kernel of spike-frequency adaptation (sfa), and the neuron
+  model can have arbitrary number of them.
+  Dynamic of each gamma_i is described by:
+
+  Tau_gamma_i*d{gamma_i}/dt = -gamma_i
+
+  and in case of spike emission, its value increased by a constant (which can be
+  positive or negative):
+
+  gamma_i = gamma_i + q_gamma_i  (in case of spike emission).
+
+
+  Parameters:
+  The following parameters can be set in the status dictionary.
+
+  Membrane Parameters:
+    C_m        double - Capacity of the membrane in pF
+    t_ref      double - Duration of refractory period in ms.
+    V_reset    double - Reset value after a spike in mV.
+    E_L        double - Leak reversal potential in mV.
+    g_L        double - Leak conductance in nS.
+    I_e        double - Constant external input current in pA.
+
+  Spike adaptation and firing intensity parameters:
+    q_stc      vector of double - Values added to spike triggered currents (stc)
+  after each spike emission in nA.
+    tau_stc    vector of double - Time constants of stc variables in ms.
+    q_sfa      vector of double - Values added to spike-frequency adaptation
+  (sfa) after each spike emission in mV.
+    tau_sfa    vector of double - Time constants of sfa variables in ms.
+    Delta_V    double - Stochasticity level in mV.
+    lambda_0   double - Stochastic intensity at firing threshold V_T in 1/s.
+    V_T_star   double - Minimum threshold in mV
+
+  Synaptic parameters
+    tau_syn_ex double - Time constant of the excitatory synaptic current in ms
+  (exp function).
```

Comment:
Do we need to mention here that it is an exponential function? It should be in the model description above.